conans/client/graph/graph_builder.py

```diff
@@ -395,20 +398,38 @@ def _resolve_recipe(self, current_node, dep_graph, requirement, check_updates,
         return new_ref, dep_conanfile, recipe_status, remote, locked_id
 
     def _create_new_node(self, current_node, dep_graph, requirement, check_updates,
-                         update, remotes, profile, graph_lock, context):
+                         update, remotes, profile_host, profile_build, graph_lock, context_switch):
+        # If there is a context_switch, it is because it is a BR-build
+        if context_switch:
+            profile = profile_build
+            context = CONTEXT_BUILD
+        else:
+            profile = profile_host if current_node.context == CONTEXT_HOST else profile_build
+            context = current_node.context
 
         result = self._resolve_recipe(current_node, dep_graph, requirement, check_updates, update,
                                       remotes, profile, graph_lock)
         new_ref, dep_conanfile, recipe_status, remote, locked_id = result
 
+        # Assign the profiles depending on the context
+        if profile_build:  # Keep existing behavior (and conanfile members) if no profile_build
+            dep_conanfile.settings_build = profile_build.processed_settings.copy()
+            if not context_switch:
+                dep_conanfile.settings_target = current_node.conanfile.settings_target
```

Comment:
Probably is worth to define ``settings_build`` and ``settings_target`` default to None for all conanfiles (in constructor), to allow recipes to quickly check ``if self.settings_target``, instead of ``if getattr(self, "settings_target")`` ?

Reply:
This feature shouldn't touch at all existing behavior without `profile_build`. Recipes in the `host` context should have:
 * `settings_target` equals `None` if using the new xbuild approach
 * no `settings_target` attribute if not using the new build.
...and same for `settings_build`.

---

This makes me think there is an error in this PR:

```python
root_node.conanfile.settings_build = profile_build.processed_settings.copy() \
    if profile_build else None
root_node.conanfile.settings_target = None
```

And I need to reimplement the PR #6778 ⚠️ 

---

I'd really prefer to initialize it to `None`, it'd be much easier, but then #6778 (re-implemented) wouldn't have all the information available.

Reply:
Maybe using ``None`` as an ``non-initialized`` value, then use ``False`` to indicate that such ``target_settings``, ``build_settings`` is not active could simplify the code?

I think that having new attributes with ``None`` as value shouldn't be changing behavior for existing code, and we could be ok with it.

Reply:
Won't be set, better not to have the attribute if no ``--profile:build`` is given.